cypress/support/step_definitions/validations.js

```diff
@@ -177,30 +177,37 @@ Then(/'(.+)' will (be|stay) in '(.+)' state/, (app, condition, state) => {
   const scenarioName = cy.state().test.title;
   const moduleReqIdJson = Cypress.env(CONSTANTS.MODULEREQIDJSON);
   const featureFileName = cy.state().test.parent.title;
-  const scenarioList = moduleReqIdJson.scenarioNames[featureFileName];
-  const validationObject = scenarioList[scenarioName].validationObject;
+  const scenarioList = moduleReqIdJson?.scenarioNames[featureFileName];
+  const validationObject = scenarioList[scenarioName]?.validationObject;
   // custom validation in case of lifecycle test cases where app is not reachable
   // if validationObject is present in the modReqId for the specific TC, we have to validate based on that value
-  if (validationObject) {
-    if (Cypress.env(CONSTANTS.COMBINEVALIDATIONOBJECTSJSON).hasOwnProperty(validationObject)) {
-      // the validation type is expected to be "custom"
-      if (
-        Cypress.env(CONSTANTS.COMBINEVALIDATIONOBJECTSJSON)[validationObject].data[0].type ==
-        'custom'
-      ) {
-        const validationObjectData = Cypress.env(CONSTANTS.COMBINEVALIDATIONOBJECTSJSON)[
-          validationObject
-        ].data[0];
-        // passing the validationObject to perform customValidation
-        cy.customValidation(validationObjectData);
-      } else {
-        assert(
-          false,
-          `Expected validationObject to be of "custom" type. Current value : ${Cypress.env(CONSTANTS.COMBINEVALIDATIONOBJECTSJSON)[validationObject].data[0].type}`
-        );
+  try {
+    if (validationObject) {
+      if (Cypress.env(CONSTANTS.COMBINEVALIDATIONOBJECTSJSON).hasOwnProperty(validationObject)) {
+        // the validation type is expected to be "custom"
+        if (
+          Cypress.env(CONSTANTS.COMBINEVALIDATIONOBJECTSJSON)[validationObject]?.data[0]?.type ==
+          'custom'
+        ) {
+          const validationObjectData = Cypress.env(CONSTANTS.COMBINEVALIDATIONOBJECTSJSON)[
+            validationObject
+          ].data[0];
+          // passing the validationObject to perform customValidation
+          cy.customValidation(validationObjectData);
+        } else {
+          assert(
+            false,
+            `Expected validationObject to be of "custom" type. Current value : ${Cypress.env(CONSTANTS.COMBINEVALIDATIONOBJECTSJSON)[validationObject].data[0].type}`
+          );
+        }
       }
     }
-  } else {
+  } catch (error) {
+    throw new Error(
+      `Following error occurred during custom validation: ${JSON.stringify(error.message)}`
+    );
+  }
```

Comment:
i think we should only print error here, right? suppose if you get an error and there is no message field?